src/libraries/System.Diagnostics.Process/ref/System.Diagnostics.Process.cs

```diff
@@ -114,13 +113,13 @@ public partial class Process : System.ComponentModel.Component
         protected void OnExited() { }
         public void Refresh() { }
         public bool Start() { throw null; }
-        public static System.Diagnostics.Process Start(System.Diagnostics.ProcessStartInfo startInfo) { throw null; }
-        public static System.Diagnostics.Process Start(string fileName) { throw null; }
-        public static System.Diagnostics.Process Start(string fileName, string arguments) { throw null; }
+        public static System.Diagnostics.Process? Start(System.Diagnostics.ProcessStartInfo startInfo) { throw null; }
+        public static System.Diagnostics.Process? Start(string fileName) { throw null; }
+        public static System.Diagnostics.Process? Start(string fileName, string arguments) { throw null; }
```

Comment:
This is one of those cases where the annotation is correct but it's probably going to cause more harm than good I think.  Still, we should keep what you have.

Reply:
It looks like the only code path that could result in a null Process value is [this one](https://github.com/dotnet/runtime/blob/852fa33030065b5367722756899004e092f704ad/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Win32.cs#L112). It doesn't seem possible on the unix implementation. 

The particular result stems from [this shell32 function](https://docs.microsoft.com/en-us/windows/win32/api/shellapi/nf-shellapi-shellexecuteexa). I don't understand enough about that particular API to judge the probability of this happening, but we do seem to throw beforehand if the call is not successful.

That being said, the nullity of the particular call seems to merely reflect the result of the also public `bool Process.Start()` instance method, so it might make sense to keep for the sake of symmetry?

Reply:
It is possible for Start to return false, if shell execute is used and returns successfully a zero handle.  However, the default for ProcessStartInfo.UseShellExecute in .NET Core is false, which means while to be accurate we do need `Start(ProcessStartInfo)` to return `Process?`, the other `Start(string)` and `Start(string, string)` overloads should never return null, because they don't use shell execute, so unless I'm missing something, those should return just `Process`.  Reasonable?

Reply:
I think your analysis makes sense. One counterpoint here is that most real uses of `Process.Start()` I've seen typically either use `ProcessStartInfo` or call the `Start()` instance method directly. So the question here is whether we want to assert non-nullability for a special case of the api based on an invariant which may change in the future or not hold in future platforms.

Reply:
> So the question here is whether we want to assert non-nullability for a special case of the api based on an invariant which may change in the future or not hold at all in future platforms.

It'd be a breaking change to start returning null in cases where we don't today, regardless of annotation, and even if documented to say null is possible.  So I'm not worried about this.  If we wanted to make such a breaking change, we'd break the annotation as part of it.

Reply:
That's fair enough